### ISSUE TITLE
ArticleUpdateWorker: updateArticles() exception IllegalArgumentException: Not a Conscrypt trust manager: android.security.net.config.RootTrustManager

### ISSUE THREAD

**Issue by Strubbl** [wallabag/android-app#983]
# Issue details

### Duplicate?
Have you searched the issues of this repository if your issue is already known? yes/

### Actual behaviour
```
2020-05-19 13:15:38.901 13971-14060/? D/DbConnection: using existing db session
2020-05-19 13:15:38.905 13971-13971/? D/greenDAO: Built SQL for query: SELECT T."_id",T."ARTICLE_ID",T."TITLE",T."DOMAIN",T."URL",T."GIVEN_URL",T."ORIGIN_URL",T."ESTIMATED_READING_TIME",T."LANGUAGE",T."PREVIEW_PICTURE_URL",T."AUTHORS",T."FAVORITE",T."ARCHIVE",T."CREATION_DATE",T."UPDATE_DATE",T."PUBLISHED_AT",T."STARRED_AT",T."IS_PUBLIC",T."PUBLIC_UID",T."ARTICLE_PROGRESS",T."IMAGES_DOWNLOADED" FROM "ARTICLE" T  WHERE T."ARTICLE_ID" IS NOT NULL AND T."ARCHIVE"=? ORDER BY T.'ARTICLE_ID' DESC LIMIT ?
2020-05-19 13:15:38.905 13971-13971/? D/greenDAO: Values for query: [0, 30]
2020-05-19 13:15:38.920 13971-13971/? D/greenDAO: Built SQL for query: SELECT T."_id",T."ARTICLE_ID",T."TITLE",T."DOMAIN",T."URL",T."GIVEN_URL",T."ORIGIN_URL",T."ESTIMATED_READING_TIME",T."LANGUAGE",T."PREVIEW_PICTURE_URL",T."AUTHORS",T."FAVORITE",T."ARCHIVE",T."CREATION_DATE",T."UPDATE_DATE",T."PUBLISHED_AT",T."STARRED_AT",T."IS_PUBLIC",T."PUBLIC_UID",T."ARTICLE_PROGRESS",T."IMAGES_DOWNLOADED" FROM "ARTICLE" T  WHERE T."ARTICLE_ID" IS NOT NULL AND T."FAVORITE"=? ORDER BY T.'ARTICLE_ID' DESC LIMIT ?
2020-05-19 13:15:38.920 13971-13971/? D/greenDAO: Values for query: [1, 30]
2020-05-19 13:15:38.931 13971-13971/fr.gaulupeau.apps.InThePoche.debug D/greenDAO: Built SQL for query: SELECT T."_id",T."ARTICLE_ID",T."TITLE",T."DOMAIN",T."URL",T."GIVEN_URL",T."ORIGIN_URL",T."ESTIMATED_READING_TIME",T."LANGUAGE",T."PREVIEW_PICTURE_URL",T."AUTHORS",T."FAVORITE",T."ARCHIVE",T."CREATION_DATE",T."UPDATE_DATE",T."PUBLISHED_AT",T."STARRED_AT",T."IS_PUBLIC",T."PUBLIC_UID",T."ARTICLE_PROGRESS",T."IMAGES_DOWNLOADED" FROM "ARTICLE" T  WHERE T."ARTICLE_ID" IS NOT NULL AND T."ARCHIVE"=? ORDER BY T.'ARTICLE_ID' DESC LIMIT ?
2020-05-19 13:15:38.931 13971-13971/fr.gaulupeau.apps.InThePoche.debug D/greenDAO: Values for query: [1, 30]
2020-05-19 13:15:38.971 13971-13971/fr.gaulupeau.apps.InThePoche.debug D/ViewRootImpl@42593f4[MainActivity]: Relayout returned: old=(0,0,1440,2960) new=(0,0,1440,2960) req=(1440,2960)0 dur=10 res=0x7 s={true 517158830080} ch=true
2020-05-19 13:15:38.972 13971-14056/fr.gaulupeau.apps.InThePoche.debug D/OpenGLRenderer: createReliableSurface : 0x77da4d81c0, 0x7869118000
2020-05-19 13:15:39.003 13971-14056/fr.gaulupeau.apps.InThePoche.debug D/mali_winsys: EGLint new_window_surface(egl_winsys_display *, void *, EGLSurface, EGLConfig, egl_winsys_surface **, EGLBoolean) returns 0x3000
2020-05-19 13:15:39.055 13971-14060/fr.gaulupeau.apps.InThePoche.debug D/NetworkSecurityConfig: Using Network Security Config from resource network_security_config debugBuild: true
2020-05-19 13:15:39.061 13971-14060/fr.gaulupeau.apps.InThePoche.debug E/ArticleUpdateWorker: updateArticles() exception
    java.lang.IllegalArgumentException: Not a Conscrypt trust manager: android.security.net.config.RootTrustManager
        at org.conscrypt.Conscrypt.toConscrypt(Conscrypt.java:736)
        at org.conscrypt.Conscrypt.setHostnameVerifier(Conscrypt.java:769)
        at okhttp3.internal.platform.ConscryptPlatform.platformTrustManager(ConscryptPlatform.kt:53)
        at okhttp3.OkHttpClient.<init>(OkHttpClient.kt:237)
        at okhttp3.OkHttpClient$Builder.build(OkHttpClient.kt:1069)
        at fr.gaulupeau.apps.Poche.network.WallabagConnection.createClient(WallabagConnection.java:123)
        at fr.gaulupeau.apps.Poche.network.WallabagConnection.createWallabagService(WallabagConnection.java:107)
        at fr.gaulupeau.apps.Poche.network.WallabagConnection.getWallabagService(WallabagConnection.java:38)
        at fr.gaulupeau.apps.Poche.service.workers.BaseNetworkWorker.getWallabagService(BaseNetworkWorker.java:86)
        at fr.gaulupeau.apps.Poche.service.workers.ArticleUpdateWorker.updateArticles(ArticleUpdateWorker.java:76)
        at fr.gaulupeau.apps.Poche.service.workers.ArticleUpdateWorker.update(ArticleUpdateWorker.java:37)
        at fr.gaulupeau.apps.Poche.service.tasks.UpdateArticlesTask.run(UpdateArticlesTask.java:17)
        at fr.gaulupeau.apps.Poche.service.tasks.ActionRequestTask.run(ActionRequestTask.java:25)
        at fr.gaulupeau.apps.Poche.service.-$$Lambda$Eg7UQkyPgCNRHvigqxzb-ridugA.run(Unknown Source:2)
        at fr.gaulupeau.apps.Poche.service.TaskService.run(TaskService.java:158)
        at fr.gaulupeau.apps.Poche.service.TaskService.lambda$gjfracnqY8x0_-mrxsx0oqjQmgk(Unknown Source:0)
        at fr.gaulupeau.apps.Poche.service.-$$Lambda$TaskService$gjfracnqY8x0_-mrxsx0oqjQmgk.run(Unknown Source:2)
        at java.lang.Thread.run(Thread.java:919)
2020-05-19 13:15:39.061 13971-14060/fr.gaulupeau.apps.InThePoche.debug D/ArticleUpdateWorker: updateArticles() finished
2020-05-19 13:15:39.062 13971-14060/fr.gaulupeau.apps.InThePoche.debug D/EventProcessor: onUpdateArticlesFinishedEvent() started
2020-05-19 13:15:39.062 13971-14060/fr.gaulupeau.apps.InThePoche.debug D/EventProcessor: onActionResultEvent() started
2020-05-19 13:15:39.062 13971-14060/fr.gaulupeau.apps.InThePoche.debug D/EventProcessor: onActionResultEvent() action: UPDATE_ARTICLES
2020-05-19 13:15:39.062 13971-14060/fr.gaulupeau.apps.InThePoche.debug D/EventProcessor: onActionResultEvent() result is success: false
2020-05-19 13:15:39.062 13971-14060/fr.gaulupeau.apps.InThePoche.debug D/EventProcessor: onActionResultEvent() result is not success; errorType: UNKNOWN
2020-05-19 13:15:39.062 13971-14060/fr.gaulupeau.apps.InThePoche.debug D/EventProcessor: onActionResultEvent() result message: java.lang.IllegalArgumentException: Not a Conscrypt trust manager: android.security.net.config.RootTrustManager
2020-05-19 13:15:39.070 13971-14060/fr.gaulupeau.apps.InThePoche.debug D/EventProcessor: onActionResultEvent() notification is not null; showing it
2020-05-19 13:15:39.072 13971-14060/fr.gaulupeau.apps.InThePoche.debug D/MainService: run() no more tasks; notifying that we are ready to stop
2020-05-19 13:15:39.072 13971-14060/fr.gaulupeau.apps.InThePoche.debug D/MainService: readyToStop()
```

### Expected behaviour
No update error.

# Steps to reproduce the issue
1. use debug version of app with latest commit of the former branch service_reorganization (which got merged with 89a0c98d3bfe6edbc3e436ee7d2dac78dead85ec)
2. update the git version to the latest head of master f37abc7a7b4a5c4313e31fcb293fdb6b38973ff0
3. run the app and let the auto-update process start

i did not change any certificate or TLS config server-side. 

This is of course not a normal way to update the app.
I am reporting this issue in case it is going to happen with a normal update, too. I am not sure if it is. If it is not going to happen to an app store user, we can close this issue.

# Environment details
- **wallabag app version**: master f37abc7a7b4a5c4313e31fcb293fdb6b38973ff0
- **wallabag app installation source** (e.g. Gplay, F-Droid, manual): manual
- **Android OS version**: 10
- **Android ROM** (e.g. stock, LineageOS, SlimRom,…): stock
- **Android hardware**: Samsung S9
- **wallabag server version**: 2.3.8
- **Do you have Two-Factor-Authentication enabled?**: no

**Comment by Strubbl** [comment]
When i run the wizard and click on "Connect" after filling in the details (server, user, pw), i get a crash:

```
2020-05-19 13:35:32.473 18132-18132/fr.gaulupeau.apps.InThePoche.debug D/ViewRootImpl@a070a41[ConnectionWizardActivity]: ViewPostIme pointer 0
2020-05-19 13:35:32.605 18132-18132/fr.gaulupeau.apps.InThePoche.debug D/ViewRootImpl@a070a41[ConnectionWizardActivity]: ViewPostIme pointer 1
2020-05-19 13:35:32.612 18132-18132/fr.gaulupeau.apps.InThePoche.debug D/Dialog: mIsSamsungBasicInteraction = false, isMetaDataInActivity = false
2020-05-19 13:35:32.637 18132-18132/fr.gaulupeau.apps.InThePoche.debug I/MultiWindowDecorSupport: updateCaptionType >> DecorView@c9b09da[], isFloating: true, isApplication: true, hasWindowDecorCaption: false, hasWindowControllerCallback: false
2020-05-19 13:35:32.638 18132-18132/fr.gaulupeau.apps.InThePoche.debug D/MultiWindowDecorSupport: setCaptionType = 0, DecorView = DecorView@c9b09da[]
2020-05-19 13:35:32.657 18132-18132/fr.gaulupeau.apps.InThePoche.debug D/ScrollView: initGoToTop
2020-05-19 13:35:32.665 18132-18132/fr.gaulupeau.apps.InThePoche.debug D/ScrollView: initGoToTop
2020-05-19 13:35:32.692 18132-18132/fr.gaulupeau.apps.InThePoche.debug D/ViewRootImpl@61644df[ConnectionWizardActivity]: setView = com.android.internal.policy.DecorView@c9b09da TM=true MM=false
2020-05-19 13:35:32.696 18132-18297/fr.gaulupeau.apps.InThePoche.debug D/TestConnectionTask: Detecting redirection for: https://mydomain.de
2020-05-19 13:35:32.713 18132-18297/fr.gaulupeau.apps.InThePoche.debug E/AndroidRuntime: FATAL EXCEPTION: AsyncTask #2
    Process: fr.gaulupeau.apps.InThePoche.debug, PID: 18132
    java.lang.RuntimeException: An error occurred while executing doInBackground()
        at android.os.AsyncTask$4.done(AsyncTask.java:399)
        at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
        at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
        at java.util.concurrent.FutureTask.run(FutureTask.java:271)
        at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:289)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:919)
     Caused by: java.lang.IllegalArgumentException: Not a Conscrypt trust manager: android.security.net.config.RootTrustManager
        at org.conscrypt.Conscrypt.toConscrypt(Conscrypt.java:736)
        at org.conscrypt.Conscrypt.setHostnameVerifier(Conscrypt.java:769)
        at okhttp3.internal.platform.ConscryptPlatform.platformTrustManager(ConscryptPlatform.kt:53)
        at okhttp3.OkHttpClient.<init>(OkHttpClient.kt:237)
        at okhttp3.OkHttpClient$Builder.build(OkHttpClient.kt:1069)
        at fr.gaulupeau.apps.Poche.network.WallabagConnection.createClient(WallabagConnection.java:123)
        at fr.gaulupeau.apps.Poche.network.WallabagConnection.createClient(WallabagConnection.java:119)
        at fr.gaulupeau.apps.Poche.network.tasks.TestConnectionTask.detectRedirection(TestConnectionTask.java:134)
        at fr.gaulupeau.apps.Poche.network.tasks.TestConnectionTask.doInBackground(TestConnectionTask.java:78)
        at fr.gaulupeau.apps.Poche.network.tasks.TestConnectionTask.doInBackground(TestConnectionTask.java:22)
        at android.os.AsyncTask$3.call(AsyncTask.java:378)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:289) 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641) 
        at java.lang.Thread.run(Thread.java:919) 
2020-05-19 13:35:32.730 18132-18132/fr.gaulupeau.apps.InThePoche.debug D/ViewRootImpl@61644df[ConnectionWizardActivity]: Relayout returned: old=(0,96,1440,2768) new=(36,1200,1404,1664) req=(1368,464)0 dur=26 res=0x7 s={true 514627735552} ch=true
2020-05-19 13:35:32.731 18132-18245/fr.gaulupeau.apps.InThePoche.debug D/OpenGLRenderer: createReliableSurface : 0x7755b7e3c0, 0x77d2341000
2020-05-19 13:35:32.735 18132-18245/fr.gaulupeau.apps.InThePoche.debug D/mali_winsys: EGLint new_window_surface(egl_winsys_display *, void *, EGLSurface, EGLConfig, egl_winsys_surface **, EGLBoolean) returns 0x3000
2020-05-19 13:35:32.736 18132-18132/fr.gaulupeau.apps.InThePoche.debug E/DecorView: mWindow.mActivityCurrentConfig is null
2020-05-19 13:35:32.771 18132-18132/fr.gaulupeau.apps.InThePoche.debug D/ViewRootImpl@61644df[ConnectionWizardActivity]: MSG_WINDOW_FOCUS_CHANGED 1 1
2020-05-19 13:35:32.815 18132-18132/fr.gaulupeau.apps.InThePoche.debug D/ViewRootImpl@c60f363[MainActivity]: stopped(false) old=true
2020-05-19 13:35:32.819 18132-18132/fr.gaulupeau.apps.InThePoche.debug D/ViewRootImpl@c60f363[MainActivity]: stopped(false) old=false
2020-05-19 13:35:32.820 18132-18132/fr.gaulupeau.apps.InThePoche.debug D/OperationsHelper: syncAndUpdate() started
2020-05-19 13:35:32.820 18132-18132/fr.gaulupeau.apps.InThePoche.debug D/OperationsHelper: updateArticles() started
2020-05-19 13:35:32.859 18132-18132/fr.gaulupeau.apps.InThePoche.debug D/ViewRootImpl@c60f363[MainActivity]: Relayout returned: old=(0,0,1440,2960) new=(0,0,1440,2960) req=(1440,2960)0 dur=8 res=0x7 s={true 514628546560} ch=true
2020-05-19 13:35:32.861 18132-18245/fr.gaulupeau.apps.InThePoche.debug D/OpenGLRenderer: createReliableSurface : 0x7749742cc0, 0x77d2407000
2020-05-19 13:35:32.862 18132-18245/fr.gaulupeau.apps.InThePoche.debug D/mali_winsys: EGLint new_window_surface(egl_winsys_display *, void *, EGLSurface, EGLConfig, egl_winsys_surface **, EGLBoolean) returns 0x3000
2020-05-19 13:35:32.899 18132-18132/fr.gaulupeau.apps.InThePoche.debug D/ViewRootImpl@a070a41[ConnectionWizardActivity]: MSG_WINDOW_FOCUS_CHANGED 0 1
2020-05-19 13:35:32.900 18132-18132/fr.gaulupeau.apps.InThePoche.debug D/InputMethodManager: prepareNavigationBarInfo() DecorView@9657bb9[ConnectionWizardActivity]
2020-05-19 13:35:32.900 18132-18132/fr.gaulupeau.apps.InThePoche.debug D/InputMethodManager: getNavigationBarColor() -14606047
2020-05-19 13:35:32.910 18132-18132/fr.gaulupeau.apps.InThePoche.debug D/MainService: onCreate()
```

**Comment by Strubbl** [comment]
I tried clearing the app's data and re-run the wizard, but same errors happen. Even after uninstall and new install via Android studio i get errors:

```
2020-05-19 13:42:49.547 22215-22375/fr.gaulupeau.apps.InThePoche.debug D/NetworkSecurityConfig: Using Network Security Config from resource network_security_config debugBuild: true
2020-05-19 13:42:49.551 22215-22375/fr.gaulupeau.apps.InThePoche.debug E/AndroidRuntime: FATAL EXCEPTION: AsyncTask #1
    Process: fr.gaulupeau.apps.InThePoche.debug, PID: 22215
    java.lang.RuntimeException: An error occurred while executing doInBackground()
        at android.os.AsyncTask$4.done(AsyncTask.java:399)
        at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:383)
        at java.util.concurrent.FutureTask.setException(FutureTask.java:252)
        at java.util.concurrent.FutureTask.run(FutureTask.java:271)
        at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:289)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:919)
     Caused by: java.lang.IllegalArgumentException: Not a Conscrypt trust manager: android.security.net.config.RootTrustManager
        at org.conscrypt.Conscrypt.toConscrypt(Conscrypt.java:736)
        at org.conscrypt.Conscrypt.setHostnameVerifier(Conscrypt.java:769)
        at okhttp3.internal.platform.ConscryptPlatform.platformTrustManager(ConscryptPlatform.kt:53)
        at okhttp3.OkHttpClient.<init>(OkHttpClient.kt:237)
        at okhttp3.OkHttpClient$Builder.build(OkHttpClient.kt:1069)
        at fr.gaulupeau.apps.Poche.network.WallabagConnection.createClient(WallabagConnection.java:123)
        at fr.gaulupeau.apps.Poche.network.WallabagConnection.createClient(WallabagConnection.java:119)
        at fr.gaulupeau.apps.Poche.network.tasks.TestConnectionTask.doInBackground(TestConnectionTask.java:107)
        at fr.gaulupeau.apps.Poche.network.tasks.TestConnectionTask.doInBackground(TestConnectionTask.java:22)
        at android.os.AsyncTask$3.call(AsyncTask.java:378)
        at java.util.concurrent.FutureTask.run(FutureTask.java:266)
        at android.os.AsyncTask$SerialExecutor$1.run(AsyncTask.java:289) 
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167) 
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641) 
        at java.lang.Thread.run(Thread.java:919) 
```

and

```
2020-05-19 13:42:49.897 22215-22215/fr.gaulupeau.apps.InThePoche.debug E/WindowManager: android.view.WindowLeaked: Activity fr.gaulupeau.apps.Poche.ui.MainActivity has leaked window DecorView@5ca9cd1[MainActivity] that was originally added here
        at android.view.ViewRootImpl.<init>(ViewRootImpl.java:765)
        at android.view.WindowManagerGlobal.addView(WindowManagerGlobal.java:429)
        at android.view.WindowManagerImpl.addView(WindowManagerImpl.java:95)
        at android.app.Dialog.show(Dialog.java:473)
        at fr.gaulupeau.apps.Poche.ui.preferences.ConfigurationTestHelper.test(ConfigurationTestHelper.java:107)
        at fr.gaulupeau.apps.Poche.ui.MainActivity.testConfiguration(MainActivity.java:913)
        at fr.gaulupeau.apps.Poche.ui.MainActivity.lambda$onResume$0$MainActivity(MainActivity.java:313)
        at fr.gaulupeau.apps.Poche.ui.-$$Lambda$MainActivity$Y174orNyQMJgrvBchecuQQEADRk.onClick(Unknown Source:2)
        at androidx.appcompat.app.AlertController$ButtonHandler.handleMessage(AlertController.java:167)
        at android.os.Handler.dispatchMessage(Handler.java:107)
        at android.os.Looper.loop(Looper.java:237)
        at android.app.ActivityThread.main(ActivityThread.java:7857)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1076)
```

**Comment by Strubbl** [comment]
the updates of okhttp after 4.5.0 are the problem

**Comment by Strubbl** [comment]
upstream issue with request for help: https://github.com/square/okhttp/issues/6069

**Comment by Strubbl** [comment]
fixed with #984
